config/webpack.config.prod.js

```diff
@@ -40,116 +33,99 @@ module.exports = {
-            // removal of unnecessary prefixes when Uglify plugin is enabled.
-            // Webpack 1.x uses Uglify plugin as a signal to minify *all* the assets
-            // including CSS. This is confusing and will be removed in Webpack 2:
-            // https://github.com/webpack/webpack/issues/283
-            autoprefixer: false,
-            // enable css modules: https://github.com/css-modules/css-modules
-            modules: true,
-            localIdentName: '[name]__[local]___[hash:base64:5]',
-            importLoaders: 1,
-          },
-        }, {
-          loader: 'postcss',
-        }]
-      )),
-    }, {
-      test: /\.json$/,
-      loader: 'json',
+      loader: 'eslint-loader',
     }, {
-      test: /\.(ico|jpg|png|gif|eot|otf|webp|svg|ttf|woff|woff2)(\?.*)?$/,
-      exclude: /(\/favicon.ico|\/icons\/.+\.svg)$/,
-      loader: 'file',
-      query: {
-        name: 'static/media/[name].[hash:8].[ext]',
-      },
-    }, {
-      test: /icons\/.+\.svg$/,
-      include: [paths.componentSrc],
-      loaders: [
-        'raw',
-        'svgo',
-      ],
-    }, {
-      test: /\/favicon.ico$/,
-      include: [paths.appSrc],
-      loader: 'file',
-      query: {
-        name: 'favicon.ico?[hash:8]',
-      },
-    }, {
-      test: /\.(mp4|webm)(\?.*)?$/,
-      loader: 'url',
-      query: {
-        limit: 10000,
-        name: 'static/media/[name].[hash:8].[ext]',
-      },
-    }, {
-      test: /\.html$/,
-      loader: 'html',
-      query: {
-        attrs: ['link:href'],
-      },
-    }, {
-      test: /\.md$/,
-      loader: 'raw',
+      oneOf: [{
+        test: /\.css$/,
+        use: ExtractTextPlugin.extract({
+          use: [{
+            loader: require.resolve('css-loader'),
+            options: {
+              autoprefixer: false,
+              modules: true,
+              localIdentName: '[name]__[local]___[hash:base64:5]',
+            },
+          }, {
+            loader: require.resolve('postcss-loader'),
+            options: {
+              config: {
+                path: path.resolve('./config/postcss.config.js'),
+              },
+            },
+          }],
+          fallback: require.resolve('style-loader'),
+        }),
+      }, {
+        test: /\.(js|jsx)$/,
+        include: [
+          paths.appSrc,
+          paths.componentSrc,
+          paths.utilsSrc,
+          paths.globalsSrc,
+          paths.constantsSrc,
+        ],
+        loader: require.resolve('babel-loader'),
+        options: babelConfig,
+      }, {
+        test: /\.html$/,
+        loader: require.resolve('html-loader'),
+        options: {
+          attrs: ['link:href'],
+        },
+      }, {
+        test: /\.json$/,
+        loader: 'json-loader',
+      }, {
+        test: /\/favicon.ico$/,
+        include: [paths.appSrc],
+        loader: 'file-loader',
+        options: {
+          name: 'favicon.ico?[hash:8]',
```

Comment:
Any specific reason why this is using query params btw? Those are bad idea generally as it invalidates all caches by default

Reply:
@mikian Correct me if I'm wrong, but I believe the hash generated is the same if the file path is the same, so the cache should be preserved unless the favicon has changed?